### 25. take a number from the user and check whether it is prime?

In [7]:

check_number = int(input("Enter a Number : "))
count = 0
for i in range(2, check_number+1):
    if check_number % i == 0 :
        count += 1
        
if count == 1:
    print(check_number, "is a Prime Number")
else:
    print(check_number, "is not a Prime Number")

Enter a Number :  4


4 is not a Prime Number


### 25. take a number from the user and check whether it is prime?

In [9]:

def is_prime_number(check_number):
    count = 0
    for i in range(2, check_number + 1):
        if check_number % i == 0:
            count += 1
    if count == 1:
        return True 
    else:
        return False 
check_number = int(input("Enter a Number: "))

if is_prime_number(check_number):
    print(check_number, "is a Prime Number")
else:
    print(check_number, "is not a Prime Number")


Enter a Number:  12


12 is not a Prime Number


### 26. take a string from the user and check contains only digits or not?

In [1]:

user_input = input("Enter a word with digits : ")
count = 0
for i in user_input:
    if not(i >= "0" and i <= "9"):
        print("contains other")
        break
    else:
        count += 1
if count == len(user_input):
    print("contains only digits")
    

Enter a word with digits :  12345


contains only digits


### 26. take a string from the user and check contains only digits or not?

In [3]:

def digits(user_input):
    return user_input.isdigit()

user_input = input("Enter a string: ")
if digits(user_input):
    print("The string contains only digits.")
else:
    print("The string contains non-digit characters.")


Enter a string:  5452kuhgfvgvbhj


The string contains non-digit characters.


### 27. take a string from the user and check contains only  alphabets or not?

In [6]:

user_input = input("Enter a word with alphabets : ")
count = 0
for i in user_input:
    if not((i >= "a" and i <= "z") or (i >= "A" and i <= "Z")) :
        print("contains others")
        break
    else:
        count += 1
if count == len(user_input):
    print("contains only alphabets")

Enter a word with alphabets :  JGJHGJGJG


contains only alphabets


### 27. take a string from the user and check contains only  alphabets or not?

In [5]:

def contains_only_alphabets(user_input):
    return user_input.isalpha()

user_input = input("Enter a string: ")
if contains_only_alphabets(user_input):
    print("The string contains only alphabets.")
else:
    print("The string contains non-alphabet characters.")


Enter a string:  12fhj#


The string contains non-alphabet characters.


### 28. take a string from the user and check contains only  special chars or not?

In [8]:

user_input = input("Enter a word with Specail chars : ")
count = 0
for i in user_input:
    if not( (i >= "a" and i <= "z") or (i >= "A" and i <= "Z") or (i >= "0" and i <= "9") ) :
        print("contains others")
        break
    else:
        count += 1
if count == len(user_input):
    print("contains only Specail chars")

Enter a word with Specail chars :  @


contains others


### 29.take a string from the user and check contains only  capital letters or not?

In [2]:

user_input = input("Enter a word with alphabets : ")
count = 0
for i in user_input:
    if not((i >= "A" and i <= "Z")) :
        print("contains others")
        
        break
    else:
        count += 1
if count == len(user_input):
    print("contains only Capital Letters")

Enter a word with alphabets :  dfhgfdjfAKJJhj


contains others


### 30.take a string from the user and check contains only  small letters or not?

In [4]:

user_input = input("Enter a word with alphabets : ")
count = 0
for i in user_input:
    if not((i >= "a" and i <= "z")) :
        print("contains others")
        break
    else:
        count += 1
if count == len(user_input):
    print("contains only small letters")

Enter a word with alphabets :  HGFGFhdfhfd


contains others


### 31. WAP to replace last n occurrence of give string.
 For example:”apple,orange,apple,grape,orange,apple,apple,orange”
 source: “apple”
 last occurrences: 2
 replace with: APPLE
 output:”apple,orange,apple,grape,orange,APPLE,APPLE,orange”

In [27]:

a = "apple,orange,apple,grape,orange,apple,apple,orange"
print(a)
list_a = a.split(',')
list_a = list_a[::-1]
source = "apple"
n = 2
count = 0
for i in range(0, len(list_a)):
    if list_a[i] == source :
        list_a[i] = "APPLE"
        count += 1
        if count == n:
            break
print(",".join(list_a[::-1]))

apple,orange,apple,grape,orange,apple,apple,orange
apple,orange,apple,grape,orange,APPLE,APPLE,orange


### 32. WAP to check given string contains numbers or not. it should consider float numbers also.

In [42]:

user_input = input("Enter a string it contains numbers and float values : " )
for i in user_input:
    if (i >= "0" and i <= "9"):
        print("contains numbers ")
        break
else:
    print("not contains numbers ")

Enter a string it contains numbers and float values :  hello


not contains numbers 


### 33. Convert the total string in to lower case. Without using lower() function.

In [45]:

user_input = input("Enter a word with alphabets : ")
user_input = list(user_input)
for i in range(0, len(user_input)):
    if ((user_input[i] >= "A" and user_input[i] <= "Z")) :
        user_input[i] = chr(ord(user_input[i]) + 32)

print("".join(user_input))

Enter a word with alphabets :  ABCabc


abcabc


### 34. Convert the total string in to upper case. Without using upper() function.

In [44]:

user_input = input("Enter a word with alphabets : ")
user_input = list(user_input)
for i in range(0, len((user_input))):
    if ((user_input[i] >= "a" and user_input[i] <= "z")) :
        user_input[i] = chr(ord(user_input[i]) - 32)

print("".join(user_input))

Enter a word with alphabets :  ABCabc


ABCABC


### 35. Show the below menu to the user until and until user select quit and display corresponding os message
Menu:
1. windows
2. Linux
3. Mac
4. quit
a = "Menu: \n1. windows \n2. Linux \n3. Mac \n4. quit"
print(a)'''

In [34]:

condition = True
while condition:
    a = "Menu: \n1. windows \n2. Linux \n3. Mac \n4. quit"
    print(a)
    user_input = input("Select a number from above options : ")
    if "1" == user_input:
        print("  ***  You selected Windows  ***  ")
        condition = True
    elif "2" == user_input:
        print(" ***  You selected Linux  ***  ")
        condition = True
    elif "3" == user_input:
        print("  *** You selected Mac  ***  ")
        condition = True
    elif "4" == user_input:
        print("  *** Thank You!  ***  ")
        condition = False
    else:
        print("===>  Select from given options please  <===")
        condition = True
        


Menu: 
1. windows 
2. Linux 
3. Mac 
4. quit


Select a number from above options :  5


===>  Select from given options please  <===
Menu: 
1. windows 
2. Linux 
3. Mac 
4. quit


Select a number from above options :  4


  *** Thank You!  ***  


### 36. take a string from the user and check contains at least one digit or not?

In [3]:

user_input = input("Enter a string it contains numbers : " )
for i in user_input:
    if (i >= "0" and i <= "9"):
        print("contains numbers ")
        break
else:
    print("not contains numbers ")

Enter a string it contains numbers :  hello


not contains numbers 


### 37. take a string from the user and check contains at least one alphabets or not?

In [6]:

user_input = input("Enter a string it contains alphabets : " )
for i in user_input:
    if (i >= "A" and i <= "Z") or (i >= "a" and i <= "z"):
        print("Contains Alphabets ")
        break
else:
    print("Not contains Alphabets ")

Enter a string it contains alphabets :  1254A


Contains Alphabets 


### 38. take a string from the user and check contains at least one chars or not?

In [9]:

user_input = input("Enter a string it contains Chars : " )
for i in user_input:
    if not ((i >= "A" and i <= "Z") or (i >= "a" and i <= "z") or (i >= "0" and i <= "9")):
        print("Contains Chars")
        break
else:
    print("Not contains chars ")

Enter a string it contains Chars :  dsggjsdgs$


Contains Chars


### 39. take a string from the user and check contains at least one capital letter or not?

In [12]:

user_input = input("Enter a string it contains alphabets : " )
for i in user_input:
    if (i >= "A" and i <= "Z"):
        print("Contains capital letters ")
        break
else:
    print("Not contains capital letters ")

Enter a string it contains alphabets :  hgfadsG


Contains capital letters 


### 40. take a string from the user and check contains at least one small letter or not?

In [16]:

user_input = input("Enter a string it contains alphabets : " )
for i in user_input:
    if (i >= "a" and i <= "z"):
        print("Contains small letters ")
        break
else:
    print("Not contains small letters ")

Enter a string it contains alphabets :  FHGCJHC


Not contains small letters 


### 41. Print the first 100 odd numbers

In [35]:

for i in range(1, 201):
    if i % 2 != 0:
        print(i, end=",")
        
    

1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95,97,99,101,103,105,107,109,111,113,115,117,119,121,123,125,127,129,131,133,135,137,139,141,143,145,147,149,151,153,155,157,159,161,163,165,167,169,171,173,175,177,179,181,183,185,187,189,191,193,195,197,199,

### 42. Determine the factors of a number entered  by the user

In [21]:

user_input = input("enter a number : ")
user_input = int(user_input)

for i in range(1, user_input+1):
    if user_input % i == 0:
        print(i, end= " ")


enter a number :  50


1 2 5 10 25 50 

### 43. Play a number guessing game (User enters a guess, you print YES or Higher or Lower).
 This should continue until and until user gives a correct number or want to quit in the middle.
 Get a hidden number by using random.randint(1,100)


In [29]:

number = random.randint(1, 100)
# print(number)
condition = True
while condition:
    user_input = input("enter a number for guess : ")
    user_input = int(user_input)
    if number == user_input:
        print("YES")
        condition = False
    elif number > user_input:
        print("Lower")
        condition = True
    else:
        print("Higher")
        condition = True

86


enter a number for guess :  90


Higher


enter a number for guess :  10


Lower


enter a number for guess :  50


Lower


enter a number for guess :  85


Lower


enter a number for guess :  88


Higher


enter a number for guess :  86


YES


### 44. Take two numbers from the user a,b check whether a is divisible by b or not?

In [56]:

a = input("enter a  number :" )
a = int(a)
b = input("enter a number : ")
b = int(a)
print(a % b)
if a % b == 0:
    print(" a is divisible by b ")
else:
    print(" a is not divisible by b ")

enter a  number : 13
enter a number :  2


0
 a is divisible by b 


### 45. Find the sum of all the multiples of 3 or 5 below 1000

In [54]:

sum_of_3 = 0
sum_of_5 = 0
for i in range(1, 1000):
    if i % 3 == 0:
        sum_of_3 += i
    if i % 5 == 0:
        sum_of_5 += i

print("sum of multiples of 3 is ", sum_of_3)
print("sum of multiples of 5 is ", sum_of_5)


sum of multiples of 3 is  166833
sum of multiples of 5 is  99500


 ### 46. Write a program to find out big of two numbers

In [2]:
user_input = input("Enter a number :")
user_input_2 = input("Enter a number :")
user_input = float(user_input)
user_input_2 = float(user_input_2)

if user_input > user_input_2 :
    print("Big number is ", int(user_input))
else:
    print("Big number is ", int(user_input_2))



Enter a number : 20
Enter a number : 30


Big number is  30


### 47. Write a program to find out biggest number in the given numbers.

In [6]:
user_input = input("Enter a Number :")
user_input = list(user_input)
big = int(user_input[0])
for i in user_input:
    if int(i) >= big:
        big = int(i)
print(big)


Enter a Number : 125321


5


### 48. find out the index of  third occurrence of given substring

In [15]:
user_input = input("Enter a String ")
user_input_2 = input("Enter a sub string")
count = 0
length = 0
list_words = user_input.split(" ")
for i in range(0, len(list_words)):
    if list_words[i] == user_input_2:
        count += 1
        if count == 3:
            break
        length += len(list_words[i])
    else:
        length += len(list_words[i])
user_input = " ".join(list_words)
if count == 3:
    print(length+count)
    

Enter a String  medi k medi l medi l
Enter a sub string medi


13


## 49. find out the index nth occurrence of given substring

In [29]:
user_input = input("Enter a String ")
user_input_2 = input("Enter a sub string")
count = 0
length = 0
list_words = user_input.split(" ")
for i in range(0, len(list_words)):
    if list_words[i] == user_input_2:
        count += 1
        length += len(list_words[i])
    else:
        length += len(list_words[i])
user_input = " ".join(list_words)

print(length+count)

Enter a String  medi medi medi
Enter a sub string medi


15


### 50. Take some single digit numbers from the user and findout min, maximum, sum, average

In [43]:
n = input("enter a number for no of inputs : ")
n = int(n)
list_a = []
for i in range(n):
    user_input = input(" enter a number ")
    list_a.append(int(user_input))
min_of = list_a[0]
max_of = list_a[0]
sum_of = 0
for i in list_a:
    sum_of += i
    if min_of >= i:
        min_of = i
    if max_of  <= i:
        max_of = i
print("mimimunm of a given numbers ", min_of)
print("Maximum of a given numbers ", max_of)
print("sum of given numbers ", sum_of)
print("Average of given numbers", (sum_of / n))

enter a number for no of inputs :  3
 enter a number  30
 enter a number  20
 enter a number  10


mimimunm of a given numbers  10
Maximum of a given numbers  30
sum of given numbers  60
Average of given numbers 20.0


### 51. print the number in proper mathematical way.
 	Consider that we have 6 digit numbers.
 Number format  WAP> 10 -> 000010
		100 ->  000100
		1000 ->  001000
		 2345678  ->  2345678
	If the number has morethan 6 digits then print as it is.

In [51]:

user_input = input("enter a number ")
length = len(user_input)
if length < 7:
    print("0" * (6- length) + user_input) #  or  #  print(user_input.zfill(6))

enter a number  321


000321


### 52. names  ="emp1,emp2,emp3,emp4" iterate through the employee names.

In [52]:
names ="emp1,emp2,emp3,emp4"
for i in names.split(","):
    print(i, input("enter emp name " ))

enter emp name  123


emp1 123


enter emp name  456


emp2 456


enter emp name  789


emp3 789


enter emp name  654


emp4 654


### 53. Take actual string, source string, destination string.
replce first nth occurrences of source string with destination string of actual string.

In [66]:

user_input = input("Enter a sentence ")
list_a = user_input.split()
print(list_a)
source = input("enter a sub string")
n = input("enter a occurances of numbers ")
n = int(n)
destination = input("enter a string to change")
count = 0
for i in range(0, len(list_a)):
    print(list_a[i])
    if list_a[i] == source :
        list_a[i] = destination
        count += 1
        if count == n:
            break
print(" ".join(list_a))

Enter a sentence  akhil m akhi m akhil m


['akhil', 'm', 'akhi', 'm', 'akhil', 'm']


enter a sub string m
enter a occurances of numbers  2
enter a string to change medi


akhil
m
akhi
m
akhil medi akhi medi akhil m


### 54. Take a two numbers from the user and do below menu driven operations
 1. addition
 2. multiples
 3.division
 4.sqrt
 5. pow    a**b
 6.subtraction
 After selection do the corresponding operation.
 Note: user may give int, or float numbers. You should check whether it is proper digits or not. 
 I.e the user given string should be in the position to convert to float. 
 Other wise show the “inproper string given” Error.


In [9]:

user_input = input("enter a number :")
user_input = float(user_input)
user_input_1 = input("enter a number :")
user_input_1 = float(user_input_1)

print("1. addition \n2. multiples \n3.division \n4.sqrt \n5. pow a**b \n6.subtraction")
select = input("select from above options : ")
if select == "1":
    print("addition of :", user_input + user_input_1)
elif select == "2":
    print("multiples  of :", user_input * user_input_1)
elif select == "3":
    print("division   of :", user_input / user_input_1)
elif select == "4":
    print("sqrt of :", user_input **2,  user_input_1**2 )
elif select == "5":
    print("pow a**b  of :", user_input ** user_input_1 )
elif select == "6":
    print("subtraction  of :", user_input - user_input_1)
else:
    print("selected inproper option")
    



enter a number : 2
enter a number : 2


1. addition 
2. multiples 
3.division 
4.sqrt 
5. pow a**b 
6.subtraction


select from above options :  5


pow a**b  of : 4.0


### 55. Take numbers from the user and find out min, maximum, sum, average

In [ ]:

n = input("enter a number for no of inputs : ")
n = int(n)
list_a = []
for i in range(n):
    user_input = input(" enter a number ")
    list_a.append(int(user_input))
min_of = list_a[0]
max_of = list_a[0]
sum_of = 0
for i in list_a:
    sum_of += i
    if min_of >= i:
        min_of = i
    if max_of  <= i:
        max_of = i
print("mimimunm of a given numbers ", min_of)
print("Maximum of a given numbers ", max_of)
print("sum of given numbers ", sum_of)
print("Average of given numbers", (sum_of / n))

### 56. l=[1,2,3,5,7,8,9,10,11,12,13,20,22,23,24,25,26,27,20,21,22,4] 
 find out how many even numbers are there and how many odd numbers are there and 
 how many positive numbers are there and how many negative numbers are there and 
 how many prime numbers are there and how many perfect numbers are there and 
 how many Armstrong numbers are there and how many palindrome numbers are there.

In [6]:

ve_no = 0
ne_no = 0
even_no = 0
odd_no = 0
prime_no = 0
perfect_no = 0
armstrong_no = 0
palindrome_no = 0

l=[1,2,3,5,7,8,9,10,11,12,13,20,22,23,24,25,26,27,20,21,22,4]
for i in l:
    if i >=0:
        ve_no += 1
    else:
        ne_no += 1
    if i % 2 == 0:
        even_no += 1
    else:
        odd_no += 1
    if i <= 1:
        prime_no += 0
    else:
        count = 0
        for j in range(2, i+1):
            if i % j == 0:
                count += 1
        if count == 1:
            prime_no += 1
    if i <= 1:
        perfect_no += 0
    else:
        sum = 1
        for j in range(2, i+1):
            if i % j == 0:
                sum += j
                if i // j != j:
                    sum += i // j
        if sum == i:
            perfect_no += 1
    num_str = str(i)
    num_digits = len(num_str)
    sum_of_powers = 0
    
    for j in num_str:
        sum_of_powers += (int(j) ** num_digits)

    if sum_of_powers == i:
        armstrong_no += 1
    if str(i) == str(i)[::-1]:
        palindrome_no += 1

print("Even numbers:", even_no)
print("Odd numbers:", odd_no)
print("Positive numbers:", ve_no)
print("Negative numbers:", ne_no)
print("Prime numbers:", prime_no)
print("Perfect numbers:", perfect_no)
print("Armstrong numbers:", armstrong_no)
print("Palindrome numbers:", palindrome_no)

Even numbers: 11
Odd numbers: 11
Positive numbers: 22
Negative numbers: 0
Prime numbers: 7
Perfect numbers: 0
Armstrong numbers: 8
Palindrome numbers: 11


### 57. Take a string from the user and find out how many digits are there, 
 how many special symbols are there, how many small letters are there, how many caps are there.

In [10]:

user_input = input("Enter a string : ")
digit_count = 0
symbol_count = 0
small_letter = 0
captial_count = 0
for i in user_input:
    if (i >= "a" and i <= "z"):
        small_letter += 1
    if (i >= "A" and i <= "Z"):
        captial_count += 1
    if (i >= "0" and i <= "9"):
        digit_count += 1
    else:
        symbol_count += 1
print("Digits:", digit_count)
print("Special symbols:", symbol_count)
print("Small letters:", small_letter)
print("Capital letters:", captial_count)


Enter a string :  Hello Akhil 123 @ aja %5jhfjdbhjb


Digits: 4
Special symbols: 29
Small letters: 20
Capital letters: 2


### 58. Take a char from the user and find out how many number of occurrences are there in given string

In [11]:

user_input = input("Enter a string :")
user_input_1 = input("Enter a char :")
count = 0
for i in user_input:
    if i == user_input_1:
        count += 1

print("no of occurances are : ", count)

Enter a string : hdgdshjgdsshsdjdoaij jjdkadskjads
Enter a char : h


no of occurances are :  3


### 59. Take a element from the user and find out how many times the  element occurred in given list

In [12]:

list_a = [1,2,3,4,5,2,5,2,6,1,2,5,2,1,5,4,3,8,9,2]
user_input = input("Enter a number :")
user_input = int(user_input)
count = 0
for i in list_a:
    if i == user_input:
        count += 1
print("no of  occurances are ", count)

Enter a number : 2


no of  occurances are  6


### 60. Take an element from the user and find out how many number of occurrences are there in given tuple

In [13]:

list_a = (1,2,3,4,5,2,5,2,6,1,2,5,2,1,5,4,3,8,9,2)
user_input = input("Enter a number :")
user_input = int(user_input)
count = 0
for i in list_a:
    if i == user_input:
        count += 1
print("no of  occurances are ", count)

Enter a number : 3


no of  occurances are  2


### 61. Reverse the string without effecting the special symbols. It involves three variations. Write code for three variations.
    Input: abc123,#$45def6%$^789$%^, output: $%^987%$^6fed54,#$321cba
    Input:abc123,#$45def6%$^789$%^, output: 9876fe,#$d54321%$^cba$%^
	Input: "123,#$456%$^789$%^", Output: 321,#$654%$^987$%^   Only numbers has to reverse.

In [1]:

s = "abc123,#$45def6%$^789$%^"
print(s)
string_reverse = s[::-1]

res = ""
sub = ""
for i in string_reverse:
    if i.isalnum():
        res += sub + i
        sub = ""
    else:
        sub = i + sub
    

print(res)

print("$%^987%$^6fed54,#$321cba", "required output" )

abc123,#$45def6%$^789$%^
$%^987%$^6fed54,#$321cba
$%^987%$^6fed54,#$321cba required output


In [3]:
# 62. Reverse the string without effecting the special symbols. It involves three variations. Write code for three variations.
#     Input:abc123,#$45def6%$^789$%^, output: 9876fe,#$d54321%$^cba$%^

s      = "abc123,#$45def6%$^789$%^"
output = "9876fe,#$d54321%$^cba$%^"


### 65. define a function to take person details name and age are mandatory parameters and height weight are optional parameters. If the user willing to pass any other details(like adhar, cell, pan, passport etc..) regarding him then your function should access those details.

In [23]:
def person_details(name, age, height=None, weight=None, **kwargs):
    details = {
        "Name": name,
        "Age": age,
        "Height": height,
        "Weight": weight
    }
    for key, value in kwargs.items():
        details[key] = value
    return details
print(person_details(name="Charlie", age=40, weight=69, cell="1234567890", passport="A1234567"))

{'Name': 'Charlie', 'Age': 40, 'Height': None, 'Weight': 69, 'cell': '1234567890', 'passport': 'A1234567'}


### 65 b. write a function to check given value is even or not

In [16]:
def even_not(value):
    return value%2==0
print(even_not(21))
print(even_not(20))

False
True


### 65 c. write a function to check given value is prime or not

In [38]:
def is_prime(value):
    if value <= 1:
        return False
    for i in range(2, int(value**0.5) + 1):
        if value % i == 0:
            return "not prime number"
    return "prime number"
print(is_prime(11))
print(is_prime(15))

prime number
not prime number


### 65 d. write a function to check given 2 values are divisible or not

In [76]:
def is_divisible(value1, value2):
    if value2 == 0:
        return "Division by zero is not allowed."
    return value1 % value2 ==0
#print(is_divisible(126,63))
#print(is_divisible(6,0))
#op =is_divisible()
value1 = float(input("enter value :"))
value2 = float(input("enter value :"))
op = is_divisible(value1, value2)
print(op)

enter value : 6
enter value : 2


True


### 67. take a string from the user and check contains only digits or not?

In [119]:
s = input('enter some string: ')
def func(s):
    for i in s:
        if 48<=ord(i)>=57:
            print("The string contains characters.")
            break
    else:
        print('digitds')
func(s)

enter some string:  23456


digitds


### 68. take a string from the user and check contains only alphabets or not?

In [5]:
def chek_only_alphabets():
    str_2=input("enter a string")
    for i in str_2:
        
        if(i>='a' and i<='z') and (i>='0'  and i<='9'):
            print("string cantaians only alphabates")
            break
    else:
        print("sring not only cantains alphabtes ")

chek_only_alphabets()

enter a string lugt98675@#$


sring not only cantains alphabtes 


### 69. take a string from the user and check contains only special chars or not?

In [15]:
s = input('enter some string: ')
def func(s):
    for i in s:
        if(i>='a' and i<='z') or (i>='0'  and i<='9') or (i>= 'A' and  i<='Z'):
            print ("string doesnt have special characters")
            break
    else:
        print("string have only special characters")
func(s)

enter some string:  @#


string have only special characters


### 70. take a string from the user and check contains only capital letters or not? 

In [39]:
s = input('Enter some string: ')
def func(s):
    for i in s:
        if 'A' <= i <= 'Z':
            print("String has capital letters")
            break
    else:
        print("String doesn't have capital letters")
func(s)

Enter some string:  Gugv b87


String has capital letters


### 71. take a string from the user and check if it contains only small letters or not?

In [37]:
s = input('enter some string: ')
def func(s):
    for i in s:
        if 'a' <= i <= 'z':
            print ("string has small letters")
            break
    else:
        print("string doesn't have small letters")
func(s)

enter some string:  lihj9876t


string has small letters


### 72. WAP to replace last n occurrence. 

In [44]:
def replace_last_n_occurrences(text, old, new, n):
    
    parts = text.rsplit(old, n)
    
    return new.join(parts)

text = "apple banana apple orange apple banana apple"
old = "apple"
new = "grape"
n = 2

result = replace_last_n_occurrences(text, old, new, n)
print("Result:", result)


Result: apple banana apple orange grape banana grape


### 73. WAP to check given string contains numbers or not. it should consider float numbers also. 

In [9]:
def contains_numbers(input_string):
    # Iterate through each character in the string
    for char in input_string:
        # Check if the character is a digit or a decimal point
        if char.isdigit() or char == '.':
            return True
    return False

# Test the function
input_string = input("Enter a string: ")
if contains_numbers(input_string):
    print("The string contains numbers.")
else:
    print("The string does not contain any numbers.")


Enter a string:  lkm


The string does not contain any numbers.


### 74 Convert the total string in to lower case. 

### Convert the total string in to upper case. 

In [13]:
def convert_case(input_string):
    lower_case = input_string.lower()  
    upper_case = input_string.upper()  
    return lower_case, upper_case

input_string = input("Enter a string: ")
lower_case, upper_case = convert_case(input_string)

print(f"Lowercase: {lower_case}")
print(f"Uppercase: {upper_case}")

Enter a string:  pl


Lowercase: pl
Uppercase: PL


### 75. Convert every word start letter into caps. Some how title not working if it contains numbers and special symbols in the word 

In [16]:
def capitalize_words(input_string):
    result = []
    words = input_string.split()  
    for word in words:
        if word: 
            new_word = word[0].upper() + word[1:]  
            result.append(new_word)
        else:
            result.append(word) 
    return ' '.join(result)
input_string = input("Enter a string: ")
output_string = capitalize_words(input_string)
print(f"Result: {output_string}")

Enter a string:  sebg asouhfcn 


Result: Sebg Asouhfcn


 ### 76.replace last two occurrences of given source string with destination string 

This is a test. This test is just a test.


### 77.write a program to check given substring is there in actual string or not? (search should be case insensitive) 

example: act="python is a pure object oriented programing language" 

In [21]:
def check_substring(case_string, substring):
    if substring.lower() in case_string.lower():
        return True
    else:
        return False
act = "python is a pure object oriented programming language"
substring = input("Enter the substring to search: ")

if check_substring(act, substring):
    print("The substring is found.")
else:
    print("The substring is not found.")

Enter the substring to search:  A


The substring is found.


### 78.check whether “pure” is there in act or not. 

Note: Use in operator 

In [9]:
def check_pure_in_string(input_string):
    # Check if the word "pure" is in the string using the 'in' operator
    if "RAMU " in input_string:
        return True
    else:
        return False

# Test the function
act = "python is a pure object oriented programming language"

if check_pure_in_string(act):
    print('"pure" is found in the string.')
else:
    print('"pure" is not found in the string.')


"pure" is not found in the string.


### 79. l=[10,20,30,[40,50,60],70,[80,90,20]]. Convert this list as single dimensional list 

In [11]:
def dimensional_list(lst):
  flat_lst = []
  for item in lst:
    if type(item) == list :
      flat_lst.extend(dimensional_list(item))
    else:
      flat_lst.append(item)
  return flat_lst
 
l = [10, 20, 30, [40, 50, 60], 70, [80, 90, 20]]
flat_l = dimensional_list(l)
print(flat_l)

[10, 20, 30, 40, 50, 60, 70, 80, 90, 20]


### 80. input: "Google" print count of each character 

In [6]:
input_string = "Google"
char_count = {char: input_string.count(char) for char in set(input_string)}
print(char_count)


{'e': 1, 'l': 1, 'g': 1, 'G': 1, 'o': 2}


### 81. Convert n dimensional list to single dimensional list. 

In [5]:
def dime(lst):
    flat_list = []
    for i in lst:
        if isinstance(i, list):
            flat_list.extend(dime(i)) 
        else:
            flat_list.append(i)
    return flat_list
nested_list = [[1, 2, [3, 4]], [5, 6], 7]
flat_list = dime(nested_list)
print(flat_list)


[1, 2, 3, 4, 5, 6, 7]


### 82. l=[1,2,3] just make it as a string. 

In [15]:
l = [1, 2, 3]
string = ""
for i in l:
    string += str(i) 
print(string)


123


### 83. l=[1,2,3,[4,5,6],7,[8,9,10]] for single dimensional list 

### 84.l=['a','A','b','B','d','D','c','C'] WAP to find out case insensitive count and 85. case insensitive search for an element.

In [15]:
l = ['a', 'A', 'b', 'B', 'd', 'D', 'c', 'C']

def case_insensitive_count(lst, element):
    return sum(1 for x in lst if x.lower() == element.lower())

def case_insensitive_search(lst, element):
    for i, x in enumerate(lst):
        if x.lower() == element.lower():
            return i
    return -1

print(case_insensitive_count(l, 'a')) 
print(case_insensitive_search(l, 'b'))


2
2


### 86. l=['a','A','b','B','d','D','c','C']  sort the list properly 

In [18]:
l = ['a', 'A', 'b', 'B', 'd', 'D', 'c', 'C']

res = sorted(l, key=str.lower)

print(res)


['a', 'A', 'b', 'B', 'c', 'C', 'd', 'D']


### 87. find the start position of the largest block of repeated characters in a given string 

In [21]:
def find_largest_repeated_block(s):
    max_start, max_len, current_start, current_len = 0, 0, 0, 1
    for i in range(1, len(s)):
        if s[i] == s[i - 1]: current_len += 1
        else:
            if current_len > max_len: max_start, max_len = current_start, current_len
            current_start, current_len = i, 1
    return current_start if current_len > max_len else max_start

# Example usage:
s = "aaabbbccccddddd"
print(find_largest_repeated_block(s))


10


### 88. WAP to find union and intersection of lists. 

In [24]:
def union(lst1, lst2):
    return list(set(lst1) | set(lst2))

def intersection(lst1, lst2):
    return list(set(lst1) & set(lst2))

list1 = [1, 2, 3, 4]
list2 = [3, 4, 5, 6]

print("Union:", union(list1, list2))
print("Intersection:", intersection(list1, list2))


Union: [1, 2, 3, 4, 5, 6]
Intersection: [3, 4]


### 89. input: fun(5) output: [1,2,3,4,3,2,1] 

In [37]:
def fun(n):
    return list(range(1, n)) + list(range(n - 2, 0, -1))

print(fun(5))


[1, 2, 3, 4, 2, 1]


### 90. input fun('abc') output: [[],][a],[b],[c],[a,b],[b,c],[c,a],[a,b,c]] 

### 91. Remove duplicates from the list: a=[1,2,3,2,3,4,1,,3,4] 

In [55]:
a = [1, 2, 3, 2, 3, 4, 1, 3, 4]

result = []
for item in a:
    if item not in result:
        result.append(item)

print(result)


[1, 2, 3, 4]


### 92. l=['1','2','3'] get the sum of the list 

In [70]:
l = ['1','2','3','4','5']
count = 0
for i in l:
    k = int(i)
    count += k
print(count)


15


In [76]:
def sum_lst(l):
    count = 0
    for i in l:
        k = int(i)
        count += k
    return count
res = sum_lst(l)
print(res)

15


### 93. l1=[1,2,3,4] l2=[5,6,7,8] sum of two lists 

In [99]:
l1 =[1,2,3,4]
l2=[5,6,7,8]

count = 0
for i in l1+l2:
    count += i
print(count)

36


In [103]:
def sum_2lst(a,b):
    count = 0
    for i in a +b:
        count += i
    return count
res = sum_2lst(l1,l2)
print(res)

36


### 94. Find third max value of element in a list with soring and without sorting a list. 

In [118]:
# l3 = [5,9,2,7,6,1,0,19,8]
# l3.sort()

# print(l3[-3])

l = [5,9,2,7,6,1,0,19,8]
first = l[0]
second = 0
third = 0

for i in l:
    if i > first:
        third = second
        second = first
        first = i
    elif i > second and i != first:
        third = second
        second = i
    elif i > third and i != second:
        third = i

print(third)

        

8
